src/components/ToDo.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {
+  useCallback, useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+import { ToDo } from '../types/ToDo';
+import { ACTIONS, DispatchContext } from '../ToDoContext';
+
+type Props = {
+  toDo: ToDo,
+};
+
+export const ToDoItem: React.FC<Props> = ({ toDo }) => {
+  const dispatch = useContext(DispatchContext);
+  const editingLi: React.RefObject<HTMLInputElement> = useRef(null);
+  const [editingtoDo, setEditingToDo] = useState(false);
+  const [rerender, setRerender] = useState(false);
+  const [editingtoDoData, setEditingToDoData] = useState({} as ToDo);
+
+  const handleClick = useCallback(() => {
+    dispatch({ type: ACTIONS.TOGGLE, payload: toDo.id });
+    setRerender(!rerender);
```

Comment:
it looks like a dirty hack :)
you should get some value from the context (reducer state?) etc , on which your render depends

Reply:
I need to create this property and throw it through the Context between the elements, whether it is more economically to use it only inside the element?
<img width="1280" alt="Screenshot 2023-08-28 at 12 52 26" src="https://github.com/mate-academy/react_todo-app/assets/20277989/897639d0-10aa-4924-a9b1-a006859109f5">